sqlglot/dialects/tsql.py

```diff
@@ -75,6 +75,20 @@ def _parse_format(args):
     )
 
 
+def _parse_eomonth(args):
+    date = seq_get(args, 0)
+    month_lag = seq_get(args, 1)
+    unit = DATE_DELTA_INTERVAL.get("month")
+
+    if month_lag:
+        args.remove(month_lag)
```

Comment:
Why was this needed?

Reply:
This is because in validate_expression a check is performed to compare the number of arguments. As EOMONTH has two, while the LAST_DAY has one, it eould gibe an error.

https://github.com/tobymao/sqlglot/blob/9015f85b79c24ed43b0e667f9b46d20884050849/sqlglot/parser.py#L758

Reply:
I see, I think a comment would help here for future reference.

Reply:
Will do